### PR TITLE
Doc: Deprecation of @unix_only etc. in NEWS

### PR DESCRIPTION
**Pull request by PallHaraldsson**
[skip ci]

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -478,6 +478,8 @@ Deprecated or removed
   * The no-op `transpose` fallback has been deprecated. Consider introducing suitable
     `transpose` methods or calling `permutedims(x, [2,1])` ([#13171], [#17075], [#17374]).
 
+  * The macros @unix_only, @linux_only, apple_only, @bsd__only, and @windows_only have been decrecated. Instead use the functions is_unix, is_linux, is_apple, is_bsd, and is_windows. It can be helpful to use them with the new @static macro. See [Handling Operating System Variation in the manual.](http://docs.julialang.org/en/latest/manual/handling-operating-system-variation/#man-handling-operating-system-variation)
```
Comment (yuyichao) [inline NEWS.md]: 1. Quote the code.
2. Please check what's the actual deprecated macro names. 2 out of 5 are wrong.

**Comment by PallHaraldsson** [comment]
Fixes #18189

File: NEWS.md [hunk 2]
```diff
@@ -478,6 +478,8 @@ Deprecated or removed
   * The no-op `transpose` fallback has been deprecated. Consider introducing suitable
     `transpose` methods or calling `permutedims(x, [2,1])` ([#13171], [#17075], [#17374]).
 
+  * The macros `@unix_only`, `@linux_only`, `@osx_only`, and `@windows_only` (and similar without `_only` ending) have been decrecated. Instead use the functions `is_unix`, `is_linux`, `is_apple`, `is_bsd`, and `is_windows`. Note, it can be helpful to use them with the new `@static` macro. See details at [Handling Operating System Variation](http://docs.julialang.org/en/latest/manual/handling-operating-system-variation/#man-handling-operating-system-variation) section in the manual.
```
Comment (tkelman) [inline NEWS.md]: deprecated is spelled wrong

**Comment by PallHaraldsson** [comment]
Thanks @tkelman.

**Comment by ViralBShah** [comment]
Merge this for 0.5rc4?

**Comment by tkelman** [comment]
do we want this one or https://github.com/JuliaLang/julia/pull/18254 ?

**Comment by ViralBShah** [comment]
#18254 looks like the one to go with, but with the manual link as in this PR.

**Pull request closed without merging**